metricsaggregator/src/main/java/io/dockstore/metricsaggregator/client/cli/TerraMetricsSubmitter.java

```diff
@@ -0,0 +1,435 @@
+                    }
+                }
+            } catch (IOException e) {
+                LOG.error("Unable to create new CSV output file", e);
+                System.exit(FAILURE_EXIT_CODE);
+            }
+
+            logStats();
+
+            if (submitTerraMetricsCommand.isRecordSkippedExecutions()) {
+                LOG.info("View skipped executions in file {}", outputFileName);
+            }
+        } catch (IOException e) {
+            LOG.error("Unable to read input CSV file", e);
+            System.exit(FAILURE_EXIT_CODE);
+        }
+    }
+
+    private void logStats() {
+        LOG.info("Done processing {} executions from Terra. Submitted {} executions. Skipped {} executions.", numberOfExecutionsProcessed, numberOfExecutionsSubmitted, numberOfExecutionsSkipped);
+    }
+
+    private void submitWorkflowExecutions(String sourceUrl, List<CSVRecord> workflowMetricRecords, WorkflowsApi workflowsApi, ExtendedGa4GhApi extendedGa4GhApi, CSVPrinter skippedExecutionsCsvPrinter) {
+        LOG.info("Processing source_url {} for {} executions", sourceUrl, workflowMetricRecords.size());
+        numberOfExecutionsProcessed.addAndGet(workflowMetricRecords.size());
+
+        if (StringUtils.isBlank(sourceUrl)) {
+            logSkippedExecutions("", workflowMetricRecords, "Can't determine TRS ID because source_url is missing", skippedExecutionsCsvPrinter, false);
+        }
+
+        // Check to see if this source_url was skipped before
+        if (skippedSourceUrlsToReason.containsKey(sourceUrl)) {
+            logSkippedExecutions(sourceUrl, workflowMetricRecords, skippedSourceUrlsToReason.get(sourceUrl), skippedExecutionsCsvPrinter, true);
+            return;
+        }
+
+        if (!sourceUrlToSourceUrlTrsInfo.containsKey(sourceUrl)) {
+            Optional<SourceUrlTrsInfo> sourceUrlTrsInfo = calculateTrsInfoFromSourceUrl(workflowMetricRecords, sourceUrl, workflowsApi, skippedExecutionsCsvPrinter);
+            if (sourceUrlTrsInfo.isEmpty()) {
+                return;
+            } else {
+                sourceUrlToSourceUrlTrsInfo.put(sourceUrl, sourceUrlTrsInfo.get());
+            }
+        }
+
+        final SourceUrlTrsInfo sourceUrlTrsInfo = sourceUrlToSourceUrlTrsInfo.get(sourceUrl);
+        final List<RunExecution> workflowExecutionsToSubmit = workflowMetricRecords.stream()
+                .map(workflowExecution -> getTerraWorkflowExecutionFromCsvRecord(workflowExecution, sourceUrlTrsInfo.sourceUrl(), skippedExecutionsCsvPrinter))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .toList();
+        final ExecutionsRequestBody executionsRequestBody = new ExecutionsRequestBody().runExecutions(workflowExecutionsToSubmit);
+        try {
+            extendedGa4GhApi.executionMetricsPost(executionsRequestBody, Partner.TERRA.toString(), sourceUrlTrsInfo.trsId(), sourceUrlTrsInfo.version(),
+                    "Terra metrics from Q4 2023");
```

Comment:
Hard-coded and arguably not correct, unless it's the date they gave it to us (metrics are for 2022, but they were given to us in Q4 2023). If we expect them to switch to invoking the API in the future, then maybe we don't need to worry about it; just flagging.

Reply:
Good point, I think I'll make it an optional argument instead

Reply:
Made the description more generic and also added `description` as a command argument